hook_functions/getdents.c

```diff
@@ -0,0 +1,45 @@
+
+struct linux_dirent {
```

Comment:
isn't it already defined in the kernel source?

Reply:
Actually it is only defined inside fs/readdir.c([as seen here](https://elixir.bootlin.com/linux/latest/A/ident/linux_dirent)), so it's not includable. it's also declared inside syscalls.h as incomplet type.

Reply:
![image](https://user-images.githubusercontent.com/170725/142508058-546284a9-7af0-4df1-a542-c249d98517e2.png)